controllers/component_controller.go

```diff
@@ -374,23 +379,30 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 
 		containerImage := component.Spec.ContainerImage
-		isUpdated := !reflect.DeepEqual(oldCompDevfileData, hasCompDevfileData) || containerImage != component.Status.ContainerImage
+		skipGitOpsGeneration := component.Spec.SkipGitOpsResourceGeneration
+		isUpdated := !reflect.DeepEqual(oldCompDevfileData, hasCompDevfileData) || containerImage != component.Status.ContainerImage || skipGitOpsGeneration != component.Status.GitOps.ResourceGenerationSkipped
 		if isUpdated {
 			log.Info(fmt.Sprintf("The Component was updated %v", req.NamespacedName))
+			component.Status.GitOps.ResourceGenerationSkipped = skipGitOpsGeneration
 			yamlHASCompData, err := yaml.Marshal(hasCompDevfileData)
 			if err != nil {
 				log.Error(err, fmt.Sprintf("Unable to marshall the Component devfile, exiting reconcile loop %v", req.NamespacedName))
 				r.SetUpdateConditionAndUpdateCR(ctx, &component, err)
 				return ctrl.Result{}, nil
 			}
 
-			// Generate and push the gitops resources
+			// Generate and push the gitops resources, if necessary.
```

Comment:
Just a Q out of curiosity. What happens in this case?

1. skipGitOpsGeneration is false during Component create, gitops resources generated
2. skipGitOpsGeneration is updated to true during Component update, gitops resources not generated/updated

So the gitops resources generated during step 1 will still remain and be possibly be out of date after step 2. Should they remain as is in the gitops repo?

Reply:
Yes, they should remain as-is on the gitops repo in that scenario.